pkg/controller/cluster/server/config.go

```diff
@@ -81,17 +85,47 @@ func serverOptions(cluster *v1beta1.Cluster, token string) string {
 	}
 
 	if cluster.Spec.Mode != agent.VirtualNodeMode {
-		opts = opts + "disable-agent: true\negress-selector-mode: disabled\ndisable:\n- servicelb\n- traefik\n- metrics-server\n- local-storage"
+		opts = opts + "disable-agent: true\negress-selector-mode: disabled\ndisable:\n- servicelb\n- traefik\n- metrics-server\n- local-storage\n"
+	}
+
+	// Adding a check for the version here to workaround issue https://github.com/rancher/k3k/issues/477
+	// in older versions of k3s
+	if requiresNetworkPolicyDisable(cluster) {
+		opts = opts + "disable-network-policy: true\n"
 	}
-	// TODO: Add extra args to the options
 
 	return opts
 }
 
-func configSecretName(clusterName string, init bool) string {
+func ConfigSecretName(clusterName string, init bool) string {
 	if !init {
 		return controller.SafeConcatNameWithPrefix(clusterName, configName)
 	}
 
 	return controller.SafeConcatNameWithPrefix(clusterName, initConfigName)
 }
+
+func requiresNetworkPolicyDisable(cluster *v1beta1.Cluster) bool {
```

Comment:
Since the flag is called `disable-network-policy` I would just use the same name:

```suggestion
func disableNetworkPolicy(cluster *v1beta1.Cluster) bool {
```